### PR TITLE
ipn/localapi: require root or sudo+operator access for SetServeConfig

### PR DESCRIPTION
**Pull request by awly**
For an operator user, require them to be able to `sudo tailscale` to use
`tailscale serve`. This is similar to the Windows elevated token check.

Updates https://github.com/tailscale/corp/issues/15405

Updates https://github.com/tailscale/corp/issues/15405

### CODE REVIEW

File: ipn/ipnauth/ipnauth.go [hunk 1]
```diff
@@ -191,6 +192,36 @@ func (ci *ConnIdentity) IsReadonlyConn(operatorUID string, logf logger.Logf) boo
...
 
+// IsLocalAdmin reports whether the connected user is considered a "local
+// admin". This is a higher privilege level than simple write access (per
+// IsReadonlyConn).
+//
+// Local admin means:
+//
+//   - Windows: member of the Administrators group (not necessarily The
+//     Administrator user, but can elevate privileges as needed)
+//   - macOS: member of the "admin" group
```
Comment (noncombatant) [inline ipn/ipnauth/ipnauth.go:203]: I think? you can, and if so should, treat macOS and Linux the same. The admin group is special on macOS only in that it has the relevant entry in /etc/sudoers:

```
%admin          ALL = (ALL) ALL
```

In principle, macOS admins might change it to add groups and users, or whatever.

File: ipn/ipnauth/ipnauth.go [hunk 2]
```diff
@@ -202,12 +233,30 @@ func isLocalAdmin(uid string) (bool, error) {
...
 		adminGroup = "administrators"
+	case runtime.GOOS == "linux":
+		return isLocalAdminLinux(u)
 	default:
 		return false, fmt.Errorf("no system admin group found")
 	}
 	return groupmember.IsMemberOfGroup(adminGroup, u.Username)
 }
 
+func isLocalAdminLinux(u *user.User) (bool, error) {
```
Comment (noncombatant) [inline ipn/ipnauth/ipnauth.go:244]: ...and then I'd change this to something like `isLocalAdminSudo`.

File: ipn/localapi/localapi.go [hunk 3]
```diff
@@ -940,14 +940,27 @@ func (h *Handler) serveServeConfig(w http.ResponseWriter, r *http.Request) {
...
+		return nil
+	}
+	if h.CallerIsLocalAdmin {
+		return nil
 	}
-	return !h.CallerIsLocalAdmin
+	switch goos {
+	case "windows":
+		return errors.New("must be a Windows local admin to serve a path")
+	case "linux":
```
Comment (noncombatant) [inline ipn/localapi/localapi.go:956]: and "macos" here

**Review by noncombatant — APPROVED** [review]

**Comment by bradfitz** [comment]
Wait, this is running "sudo" as child process on every LocalAPI call??

**Comment by awly** [comment]
https://github.com/tailscale/tailscale/pull/10163 to avoid running sudo on every LocalAPI call

**Pull request merged**